data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/consumer/KafkaSourceCustomConsumer.java

```diff
@@ -244,15 +267,27 @@ Map<TopicPartition, OffsetAndMetadata> getOffsetsToCommit() {
 
     @Override
     public void run() {
-        consumer.subscribe(Arrays.asList(topicName));
+        consumer.subscribe(Arrays.asList(topicName), this);
+        Set<TopicPartition> partitions = consumer.assignment();
+        synchronized (ownedPartitionsEpoch) {
+            final long currentEpoch = getCurrentTimeNanos();
+            partitions.forEach((partition) -> {
+                final OffsetAndMetadata offsetAndMetadata = consumer.committed(partition);
+                LOG.info("Starting consumer with topic partition ({}) offset {}", partition, offsetAndMetadata);
+                ownedPartitionsEpoch.put(partition, currentEpoch);
+            });
+        }
+
```

Comment:
Can concurrent hashmap be used instead of this synchronized block ? concurrent hashmap will have better performance.

Reply:
ok we can address that in future.